### PR TITLE
feat: Add plugin listing to "kn --help"

### LINKED ISSUE #266: Consider adding plugin cmds to `kn --help`
## In what area(s)?

Classifications:
/kind good-first-issue
/kind feature
/kind proposal

## Describe the feature:

Once we have #249 merged, we should consider making `kn --help` show the list of available plugins.  I still think having the `plugin` command is good, but from a UX perspective one of the points of the plugin model is that the user is (for the most part) unaware that they're using a plugin vs a built-in command. So, if we can make `kn --help` show the list of plugins under some kind of `misc` section I think that would be really helpful. Even if the help text for each one just said `see kn foo for more details`, the name of the command should at least provide a hint as to what it does.  Often times people know there's a certain command they want to run but can't remember the name (get vs describe vs inspect vs ... ) so `kn --help` showing the full list of cmds will help them find what they're looking for.

### LINKED ISSUE #904: Plugin handling if it retuns with non-zero exit status
### Bug report

If a plugin exits with non-zero status, help usage hint is printed, which seems to be unnecessary.

### Expected behavior

If plugin execution path is invoked, let plugin print message or error.

### Steps to reproduce the problem
1. Create a sample plugin
```
cat ~/.config/kn/plugins/kn-source-test.sh 
#!/bin/bash
echo "hello"
exit 1
```
2. Make sure its executable 
```
chmod +x ~/.config/kn/plugins/kn-source-test.sh
```
3. Run
```
kn source test.sh
hello
Error: exit status 1
Run 'kn --help' for usage
```

Notice the last help usage message printed.

### kn version
master
```
kn version 
Version:      v20200619-local-a98d22a1-dirty
Build Date:   2020-06-19 15:35:19
Git Revision: a98d22a1
Supported APIs:
* Serving
  - serving.knative.dev/v1 (knative-serving v0.15.0)
* Eventing
  - sources.knative.dev/v1alpha2 (knative-eventing v0.15.0)
  - eventing.knative.dev/v1alpha1 (knative-eventing v0.15.0)
```
/kind cleanup

### PR DESCRIPTION
**Pull request by rhuss**
This works on all levels. Test needs to be expanded still.

Fixes #266
Fixes #904

### CODE REVIEW

File: pkg/kn/plugin/manager.go [hunk 1]
```diff
@@ -146,11 +159,29 @@ func (manager *Manager) ListPlugins() (PluginList, error) {
...
 	return plugins, nil
 }
 
+func isPartOfCommandGroup(commandGroupParts []string, name string) bool {
+	if commandGroupParts == nil {
+		return true
+	}
+
+	commandParts := extractPluginCommandFromFileName(name)
+	if len(commandParts) != len(commandGroupParts)+1 {
```
Comment (maximilien) [inline pkg/kn/plugin/manager.go:174]: Can we get some English comments for these conditions... typically not for comments but when conditions are not obvious...

Comment (rhuss) [reply pkg/kn/plugin/manager.go:174]: Sure.

**Comment by rhuss** [comment]
@navidshaikh @maximilien tests are green, I also slurped in #910 so that we can get quicker to the release.

**Comment by rhuss** [comment]
There is probably still someplace for improvement (e.g. how to get to the description of a plugin, for now just the path is printed). But maybe let's leave that for the next release.

**Pull request merged**